client/lib/media/library-selected-store.js

```diff
@@ -84,9 +84,15 @@ MediaLibrarySelectedStore.getAll = function( siteId ) {
 		return [];
 	}
 
-	return MediaLibrarySelectedStore._media[ siteId ].map( function( itemId ) {
-		return MediaStore.get( siteId, itemId );
-	} );
+	// Avoid keeping invalid items in the selected list.
+	return (
+		MediaLibrarySelectedStore
+			._media[ siteId ]
+			.map( itemId => MediaStore.get( siteId, itemId ) )
+			.filter( ( item ) => {
+				return ( item && item.guid );
```

Comment:
Thanks, misread that line there. I was mostly concerned about the transient shape not always having guid available. https://github.com/Automattic/wp-calypso/blob/master/client/lib/media/utils.js#L530-L537

I'll test the case